core/src/main/java/org/apache/calcite/sql/dialect/BigQuerySqlDialect.java

```diff
@@ -290,7 +306,7 @@ private void unparseTrim(SqlWriter writer, SqlCall call, int leftPrec,
     // If the trimmed character is a non-space character, add it to the target SQL.
     // eg: TRIM(BOTH 'A' from 'ABCD'
     // Output Query: TRIM('ABC', 'A')
-    if (!valueToTrim.toValue().matches("\\s+")) {
+    if (!valueToTrim.toValue().matches("\\s*")) {
```

Comment:
Why is this changed needed?

Reply:
Since the null byte string is empty not having this would result in TRIM(<byte>, b''). Is this valid?

Reply:
Is this different than if the argument was an empty hex char string in TRIM()?

Reply:
It is in the case of byte since by the documentation, if a byte is passed in the default is a null byte.

Reply:
Can you paste a snippet of the text from the docs that you're referring to?

Reply:
If you omit this option (refers to BOTH/TRAILING/LEADING), the default is BOTH, and the default trim character is a null byte for byte types and a pad character for character types.

Reply:
Ok I think I see what you mean. So this is checking whether the character to trim is a space character or not? Not the string to be trimmed.

Reply:
Yeah exactly.

Reply:
Ok in that case, let's just update the comment above to indicate that it should be a `non-space character and non-empty`.